packages/trader/src/Modules/Trading/Containers/__tests__/trade.spec.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { ComponentProps } from 'react';
+import { act, render, screen } from '@testing-library/react';
+import { StoreProvider, mockStore } from '@deriv/stores';
+import Trader from '../trade';
+import TraderProviders from '../../../../trader-providers';
+import { isDesktop, isMobile } from '@deriv/shared';
+import ChartLoader from 'App/Components/Elements/chart-loader';
+
+jest.mock('App/Components/Elements/PositionsDrawer', () => jest.fn(() => <div>PositionsDrawer</div>));
+jest.mock('../trade-chart', () => jest.fn(() => <div>TradeChart</div>));
+jest.mock('../../Components/Form/form-layout', () => jest.fn(() => <div>FormLayout</div>));
+jest.mock('App/Components/Elements/chart-loader', () =>
+    jest.fn(({ is_visible }: ComponentProps<typeof ChartLoader>) => (is_visible ? <div>ChartLoader</div> : null))
+);
+
+jest.mock('@deriv/shared', () => ({
+    ...jest.requireActual('@deriv/shared'),
+    isDesktop: jest.fn(),
+    isMobile: jest.fn(),
+    TRADE_TYPES: jest.fn(),
+}));
+
+const rootStore = mockStore({
+    ui: {
+        notification_messages_ui: 'div',
+        is_mobile: false,
+    },
+    modules: {
+        trade: {
+            contract_type: 'CALL',
+            form_components: [],
+            getFirstOpenMarket: jest.fn(),
+            has_barrier: false,
+            is_accumulator: false,
+            is_chart_loading: false,
+            is_market_closed: false,
+            is_synthetics_available: false,
+            is_synthetics_trading_market_available: false,
+            is_trade_enabled: true,
+            is_trade_params_expanded: false,
+            is_turbos: false,
+            is_vanilla: false,
+            onMount: jest.fn(),
+            onUnmount: jest.fn(),
+            prepareTradeStore: jest.fn(),
+            setIsDigitsWidgetActive: jest.fn(),
+            setMobileDigitView: jest.fn(),
+            should_show_active_symbols_loading: false,
+            show_digits_stats: true,
+            symbol: 'symbol',
+        },
+    },
+});
+
+const MockTrader = () => (
+    <StoreProvider store={mockStore({})}>
+        <TraderProviders store={rootStore}>
+            <Trader />
+        </TraderProviders>
+    </StoreProvider>
+);
+
+describe('Trader', () => {
+    beforeEach(() => {
```

Comment:
In order not to overwrite store values in test cases you can move `rootStore` inside of `beforeEach()` and just pass it as argument in `render(<MockTrader />)`. It will make all your changes of store values in each test case independent and they will not influence on each other. Also you could get rid of such things as  
```
        const onMount = jest.fn();
        const onUnmount = jest.fn();
```
 But there is a drawback - you'll create a store a lot of times))